src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -21,6 +20,7 @@ public class Connection: IConnection
         private IChannelConnection<Message> _transportChannel;
         private ITransport _transport;
         private Task _receiveLoopTask;
+        private Task _startTask = Task.CompletedTask;
```

Comment:
That's a very good point. I was thinking about `_startTask` a bit but not about `_connectionState`.